inc/class-core-sitemaps-renderer.php

```diff
@@ -150,7 +150,10 @@ public function get_sitemap_index_xml( $sitemaps ) {
 
 		foreach ( $sitemaps as $entry ) {
 			$sitemap = $sitemap_index->addChild( 'sitemap' );
-			$sitemap->addChild( 'loc', esc_url( $entry['loc'] ) );
+			// Add each attribute as a child node to the <sitemap> entry.
+			foreach ( $entry as $attr => $value ) {
```

Comment:
Loop, because there's no a filter for this list (e.g. to add back `lastmod` if wanted)